src/main/java/core/basesyntax/CircleSupplier.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class CircleSupplier {
+    private Random random = new Random();
+    private RandomSize size = new RandomSize();
+    private ColorSupplier color = new ColorSupplier();
+
+    public Figure getRandomCircle() {
+        return new Circle(size.getRandomSize(), color.getRandomColor());

```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables for size and color and pass them on for better code readability.